tests/tools/oesign/test-enclave/enclave/CMakeLists.txt

```diff
@@ -0,0 +1,25 @@
+# Copyright (c) Open Enclave SDK contributors.
+# Licensed under the MIT License.
+
+set(EDL_FILE ../helloworld.edl)
+
+# Use the edger8r to generate C bindings from the EDL file.
+add_custom_command(
+  OUTPUT helloworld_t.h helloworld_t.c helloworld_args.h
+  DEPENDS ${EDL_FILE} edger8r
+  COMMAND
+    edger8r --trusted ${EDL_FILE} --search-path ${PROJECT_SOURCE_DIR}/include
+    --search-path ${PLATFORM_EDL_DIR} --search-path ${CMAKE_CURRENT_SOURCE_DIR})
+
+add_executable(oesign_test_enc enc.c ${CMAKE_CURRENT_BINARY_DIR}/helloworld_t.c)
+
+if (WIN32)
+  maybe_build_using_clangw(oesign_test_enc)
+endif ()
+
+target_compile_definitions(oesign_test_enc PUBLIC OE_API_VERSION=2)
+
+# Need for the generated file helloworld_t.h
+target_include_directories(oesign_test_enc PRIVATE ${CMAKE_CURRENT_BINARY_DIR})
```

Comment:
`enclave_include_directories` and `enclave_link_libraries`

Reply:
For consistency with `add_executable`, this elides the additional overhead with the `enclave*` macros.